### DISCUSSION TITLE
Is it possible override a configuration flag?

### DISCUSSION THREAD

**Discussion by alcohol** [mlocati/docker-php-extension-installer#740] [Q&A]
When trying to install `phpredis/phpredis@develop` in my container it keeps failing because the installer explicitly assumes lzf should be enabled without any verification or reasoning. Can I override this from the command line, or do I have to patch the script?

https://github.com/mlocati/docker-php-extension-installer/blob/7655337ec9a43dfbe8e2e889c31fb6ad2e68f7e0/install-php-extensions#L3140

**Comment by mlocati** [comment]
> explicitly assumes lzf should be enabled without any verification or reasoning

In [these lines](https://github.com/mlocati/docker-php-extension-installer/blob/7655337ec9a43dfbe8e2e889c31fb6ad2e68f7e0/install-php-extensions#L3142-L3155) we make sure that the `libzstd` system library is installed.

Does `enable-redis-zstd` require that we must be sure we have the `zstd` PHP library instead? All the tests have passed so far by simply installing the `libzstd` system library...

**Reply by alcohol** [reply]
Not sure why that matters. `libzstd` has nothing to do with `lzf` afaik. My container is alpine based and does not have the packages `liblzf` nor `liblzf-dev` installed which contain the necessary files `/usr/include/lzf.h` and `/usr/include/lzfP.h`. I worked around it by patching the installer though to simply pass `no` instead of `yes` explicitly.

**Comment by mlocati** [comment]
Whoops, sorry... you wrote `lzf` but I read `zstd`... I think I need a break :wink:

I've tried to change

```sh
addConfigureOption 'enable-redis-lzf' 'yes' 
```

to

```sh
addConfigureOption 'enable-redis-lzf' 'no' 
```

But I still have the following error:

```
ERROR: file /tmp/src/tmp.ffJeGh/phpredis-develop/liblzf/LICENSE does not exist
```

I think that the reason is that the code directly downloaded from GitHub is missing the liblzf submodule, and the PHP extension assumes it's there even if it won't use lzf.

**Comment by mlocati** [comment]
If you really need to download code from github, I think that your only option is to use these commands:

```sh

# Install git on Alpine Linux
apk add git
# Install git on Debian
apt update && apt install -qy git

# Clone the repo and install submodules
git clone -b develop --depth 1 --single-branch --recurse-submodules https://github.com/phpredis/phpredis.git /tmp/phpredis

# Fix bug - see https://github.com/phpredis/phpredis/pull/2346
mv /tmp/phpredis/array.md /tmp/phpredis/arrays.md || true

# Install it
/app/install-php-extensions /tmp/phpredis

# Cleanup
rm -rf /tmp/phpredis
```

**Reply by alcohol** [reply]
Odd, I didn't have an issue with that once I changed the configuration flag to `no`. Thanks for looking into this though. It's an edge-case scenario and not really worth spending a lot of time on I figure. I merely ran into this cause I needed to create a php container with debugging symbols and the redis extension has a [bug in said context](https://github.com/phpredis/phpredis/issues/2041) that is only resolved in their develop branch so far.
